docs/ChainOrderTransitivity.md

```diff
@@ -0,0 +1,153 @@
+ - `B < C` due to the VRF tiebreaker, as the slots 3 and 6 are within `Δ = 5`, again.
+ - `C = A` as the slots 6 and 0 differ by more than `Δ = 5`.
+
+So we have `A < B < C = A < B < C = ...`.
+
+## History
+
+The relevant changes to the chain order occurred in these PRs in chronological order:
+
+ - [ouroboros-network#2108](https://github.com/IntersectMBO/ouroboros-network/pull/2108) added the opcert number comparison step as the first tiebreaker (before that, only chain length was used).
+ - [ouroboros-network#2195](https://github.com/IntersectMBO/ouroboros-network/pull/2195) added the VRF tiebreaker, but lexicographically *before* the opcert number tiebreaker, in contrast to the status quo.
+   This means that the order still was total at this point, also see below.
+ - [ouroboros-network#2348](https://github.com/IntersectMBO/ouroboros-network/pull/2348) mostly did a change unrelated to the purpose of this document, but crucially, it swapped the order in which the VRF and opcert number comparison are done, introducing the current non-transitivity.
+
+## Impact of non-transitivity
+
+Fundamentally, the Praos allows ties to be broken arbitrarily.
+Given that we always first compare by chain length, the fundamental Consensus properties (like Common Prefix) should not be affected by the non-transitivity.
+
+Still, there are some potential complications of non-transitivity:
+
+ - Implementation-related.
+    - Even though the non-transitivity gives rise to "cycles" like `A < B < C < A < B < C < ...` as seen above, it is guaranteed that we will never end up changing our selection like this, eg as blocks that already are in the VolatileDB are not added again.
+    - We currently call `sortBy` from `base` in chain selection.
+      The docs of `sortBy` state:
+
+      > The relation is also expected to be transitive: if it is not then `sortBy` might fail to find an ordered permutation, even if it exists.
+
+      According to some preliminary property tests, however, this is not the case for our particular non-transitive order, but it is certainly a weird state of affairs.
+      For example, do we have to worry that GHC changes its implementation of `sortBy` in a way that interacts non-trivially with our non-transitive order?
+    - Also, the reliance on `sortBy` is not crucial, we see several ways how to change our code such that it would do something well-defined even for non-transitive tiebreakers.
+ - Reasoning-related.
+    - Having a non-transitive order could turn out to be very confusing for reasoning about anything that is related to the chain order.
+      This is because transitivity is so ingrained when thinking about orders that it is hard to not implicitly assume.
+    - Certain "obvious" properties suddenly fail to hold, for example:
+
+      > Repeatedly observing the selection of a node over a period of time yields a sequence that is strictly improving for every two adjacent selections.
```

Comment:
I'd say "adjacent observations" instead of "adjacent selections", to emphasize the subsequence aspect of it?

Reply:
I think you'll want to use at least one of the words "subsequence" or "sample" in order to emphasize that the observations do not necessarily include every block the node selected during that time.

Reply:
Rephrased and added an example :+1: